pyincore/hazardservice.py

```diff
@@ -1061,305 +1061,22 @@ def search_hurricanewf(self, text: str, skip: int = None, limit: int = None, tim
-                    "demand_unit": [
-                        "in",
-                        "ft",
-                        "m"
-                    ],
-                    "description": "Peak ground displacement"
-                },
-                {
-                    "demand_type": "sa",
-                    "demand_unit": [
-                        "g",
-                        "in/sec^2",
-                        "m/sec^2"
-                    ],
-                    "description": "Spectral acceleration"
-                },
-                {
-                    "demand_type": "sd",
-                    "demand_unit": [
-                        "in",
-                        "ft",
-                        "m",
-                        "cm"
-                    ],
-                    "description": "Spectral displacement"
-                },
-                {
-                    "demand_type": "sv",
-                    "demand_unit": [
-                        "cm/s",
-                        "in/s"
-                    ],
-                    "description": "Spectral Velocity"
-                },
-                {
-                    "demand_type": "Hmax",
-                    "demand_unit": [
-                        "ft",
-                        "m"
-                    ],
-                    "description": "Onshore: maximum tsunami height above local ground level overland. "
-                                   "Offshore: maximum tsunami height taken crest to trough"
-                },
-                {
-                    "demand_type": "Vmax",
-                    "demand_unit": [
-                        "mph",
-                        "kph",
-                        "ft/sec",
-                        "m/sec"
-                    ],
-                    "description": "Maximum near-coast or overland water velocity due to tsunami"
-                },
-                {
-                    "demand_type": "Mmax",
-                    "demand_unit": [
-                        "m^3/s^2",
-                        "ft^3/s^2"
-                    ],
-                    "description": ""
-                }
-            ]
-        }
-
-        if hazard_type:
-            return allowed_demands[hazard_type]
-
-        return allowed_demands
+    def get_allowed_demands(self, hazard_type, timeout=(30, 600), **kwargs):
+        url = ""
+        if hazard_type == 'earthquake':
+            url = urljoin(self.base_earthquake_url, "demands")
+        elif hazard_type == 'tornado':
+            url = urljoin(self.base_tornado_url, "demands")
+        elif hazard_type == 'tsunami':
+            url = urljoin(self.base_tsunami_url, "demands")
+        elif hazard_type == 'hurricane':
+            url = urljoin(self.base_hurricane_url, "demands")
+        elif hazard_type == 'hurricaneWindfield':
+            url = urljoin(self.base_hurricanewf_url, "demands")
+        elif hazard_type == 'flood':
```

Comment:
Should we consider some kind of debug statement or exception here (e.g. unknown hazard demand type)?

Reply:
added exception